R/target_market_share.R

```diff
@@ -316,15 +315,72 @@ target_market_share <- function(data,
   data <- data %>%
     pivot_wider2()
 
+  data <- data %>%
+    left_join(green_or_brown, by = c("sector", "technology")) %>%
+    left_join(tmsr_or_smsp, by = "green_or_brown") %>%
+    rename(target_name = .data$which_metric) %>%
+    select(-.data$green_or_brown)
+
   corporate_economy <- calculate_ald_benchmark(ald, region_isos, by_company)
 
   relevant_sectors <- unique(data$sector)
 
   relevant_corporate_economy <- corporate_economy %>%
     filter(.data$sector %in% relevant_sectors)
 
+  relevant_corporate_economy <- relevant_corporate_economy %>%
+    left_join(green_or_brown, by = c("sector", "technology")) %>%
+    left_join(tmsr_or_smsp, by = "green_or_brown") %>%
+    rename(target_name = .data$which_metric) %>%
+    select(-.data$green_or_brown)
+
+  data <- data %>%
+    dplyr::bind_rows(relevant_corporate_economy)
+
+  percent_by_sector_groups <- maybe_add_name_ald(
+    c("sector", "region", "scenario_source", "metric"),
+    by_company
+  )
+
+  data <- data %>%
+    group_by(!!!rlang::syms(c(percent_by_sector_groups, "technology"))) %>%
+    arrange(.data$year) %>%
+    mutate(initial_technology_production = first(.data$production)) %>%
+    group_by(!!!rlang::syms(c(percent_by_sector_groups, "year"))) %>%
+    mutate(
+      .sector_production = sum(.data$production)
+      ) %>%
```

Comment:
why is this spread across 3 lines?

also consider if you need to use `na.rm = TRUE` in this `sum()`

Reply:
It is spread across 3 lines because I forgot to run the styler hehe >:)

Also I opted to have `target_market_share` be very picky, and fail if any crucial input columns contain `NA`s, so in principle, we shouldn't need to worry about that. If there are `NA`s, they will have been introduced by a bug in the code, and in general, I would rather know about that bug, than have it pass silently.